phone/src/apps/messages/components/ui/MessageEmbed.tsx

```diff
@@ -14,53 +14,73 @@ interface MessageEmbedProps {
   type: string;
   embed: any;
   isMine: boolean;
+  message: string;
 }
 
 type MessageEmbedType = {
   [key: string]: JSX.Element;
 };
 
-const MessageEmbed: React.FC<MessageEmbedProps> = ({ type, embed, isMine }) => {
+const MessageEmbed: React.FC<MessageEmbedProps> = ({ type, embed, isMine, message }) => {
   const embedType: MessageEmbedType = {
-    contact: <ContactEmbed embed={embed} isMine={isMine} />,
-    location: <LocationEmbed embed={embed} isMine={isMine} />,
+    contact: <ContactEmbed embed={embed} isMine={isMine} message={message} />,
+    location: <LocationEmbed embed={embed} isMine={isMine} message={message} />,
   };
 
   return <>{embedType[type]}</>;
 };
 
-const ContactEmbed = ({ isMine, embed }: { isMine: boolean; embed: Contact }) => {
+const ContactEmbed = ({
+  isMine,
+  embed,
+  message,
+}: {
+  isMine: boolean;
+  embed: Contact;
+  message: string;
+}) => {
   const [t] = useTranslation();
   const history = useHistory();
   const { pathname } = useLocation();
   const { getContactByNumber } = useContactActions();
 
   const showAddButton = !isMine && !getContactByNumber(embed?.number);
+  const showMessage = message !== t('MESSAGES.CONTACT_SHARED');
 
   const handleAddContact = () => {
     const referal = encodeURIComponent(pathname);
     history.push(`/contacts/-1?addNumber=${embed.number}&name=${embed.display}&referal=${referal}`);
   };
 
   return (
-    <StyledMessage>
-      <Box>
-        <Avatar src={embed?.avatar} />
-        <Typography>{embed?.display}</Typography>
-        <Typography>{embed?.number}</Typography>
-      </Box>
-      {showAddButton && (
+    <StyledMessage flexDirection="column">
+      {showMessage && <Box sx={{ width: '100%' }}>{message}</Box>}
+      <Box display="flex" alignItems="center" sx={{ width: '100%' }}>
         <Box>
-          <Button fullWidth variant="contained" color="primary" onClick={handleAddContact}>
-            {t('GENERIC.ADD')}
-          </Button>
+          <Avatar src={embed?.avatar} />
+          <Typography>{embed?.display}</Typography>
+          <Typography>{embed?.number}</Typography>
         </Box>
-      )}
+        {showAddButton && (
+          <Box pl={'12px'}>
```

Comment:
Hardcoded pixel values, how well does this scale?

Reply:
From what I tested it fitted well. It is the same value as the X padding of the message bubble. I took it from devtools but I can edit it if you want.

Reply:
Changed to `pl={1}` in last commit